mungegithub/mungers/lgtm_handler.go

```diff
@@ -0,0 +1,172 @@
+type LGTMHandler struct{}
+
+func init() {
+	l := LGTMHandler{}
+	RegisterMungerOrDie(l)
+}
+
+// Name is the name usable in --pr-mungers
+func (LGTMHandler) Name() string { return "lgtm-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (LGTMHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (LGTMHandler) Initialize(config *github.Config, features *features.Features) error { return nil }
+
+// EachLoop is called at the start of every munge loop
+func (LGTMHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (LGTMHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h LGTMHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
+	reviewers := getReviewers(obj)
+	if len(reviewers) == 0 {
+		return
+	}
+
+	comments, err := getComments(obj)
+	if err != nil {
+		glog.Errorf("unexpected error getting comments: %v", err)
+	}
+
+	if !obj.HasLabel(lgtmLabel) {
+		h.addLGTMIfCommented(obj, comments, reviewers)
+		return
+	}
+	h.removeLGTMIfCancelled(obj, comments, reviewers)
+}
+
+func (h *LGTMHandler) addLGTMIfCommented(obj *github.MungeObject, comments []*githubapi.IssueComment, reviewers []*githubapi.User) {
+	// Assumption: The comments should be sorted (by default from github api) from oldest to latest
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		fields := strings.Fields(*comment.Body)
+		if isCancelComment(fields) {
+			// "/lgtm cancell" if commented more recently than "/lgtm"
+			return
+		}
+		if isLGTMComment(fields) {
+			// TODO: support more complex policies for multiple reviewers.
+			// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+			// TODO: An approver should be acceptable.
+			// See https://github.com/kubernetes/contrib/pull/1428#discussion_r72563935
+			if isValidUser(comment.User) && isReviewer(comment.User, reviewers) {
+				glog.Infof("Adding lgtm label. Reviewer (%s) LGTM", *comment.User.Login)
+				obj.AddLabel(lgtmLabel)
+				return
+			}
+		}
+	}
+}
+
+func (h *LGTMHandler) removeLGTMIfCancelled(obj *github.MungeObject, comments []*githubapi.IssueComment, reviewers []*githubapi.User) {
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		fields := strings.Fields(*comment.Body)
+		if isLGTMComment(fields) {
+			// "/lgtm" if commented more recently than "/lgtm cancel"
+			return
+		}
+		if isCancelComment(fields) {
+			if isValidUser(comment.User) && isReviewer(comment.User, reviewers) {
+				glog.Infof("Removing lgtm label. Reviewer (%s) cancelled", *comment.User.Login)
+				obj.RemoveLabel(lgtmLabel)
+				return
+			}
+		}
+	}
+}
+
+func isLGTMComment(fields []string) bool {
+	return len(fields) == 1 && strings.ToLower(fields[0]) == "/lgtm"
```

Comment:
This is fine for now, but later we'd probably move all the bot-command parsing code to its own package.